File: webApi/Interface/IUser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using webApi.Models.user;

namespace webApi.Interface
{
  public  interface IUser
    {
        Task<userModels> Authenticate(string username , string password );
        void Register(string firstname, string lastname, string username,string email, string password,DateTime dob,string phonenumber);
        Task<bool> UserAlreadyExist(string username);
    }
}

File: Fonted/src/interface/Iuser.ts
export interface Iuser {
    firstname: String;
    lastname: string;
    username: string;
    email: string;
    phonenumber: string;
    image: string;
    gender: string;
    password: string;
    confirmpassword: string;
}
export interface IUserLogin{
  username: string|undefined;
  password: string|undefined;
  token:string|undefined;

}
